docs/DeveloperGuide.adoc

```diff
@@ -96,64 +112,413 @@ The `UI` component,
+
+image::CreatingTagNoteCommand.png[]
+
+**Step 2:** `NoteParserUtil` will be used to create a `noteId` object.
+
+**Step 3:** Finally, `TagParserUtil` will be used to create `tag` objects. All of them will be aggregated inside a `tagList`.
+
+This whole process has created a `TagNoteCommand` object from user input.
+
+===== Executing Tag Command
+
+Now, we will take a look on how we are executing the tagging command.
+
+Given below is an example scenario when the tagging command gets executed.
+
+**Step 1:** The `TagNoteCommand` will first exchange each tag with a tagId through model. Internally, model will have to
+interact with TagManager which will find the tag or create it if it does not exist.
+
+image::ExecutingTagNoteCommand.png[]
+
+**Step 2:** The `TagNoteCommand` then interact with `NoteManager` in order to find the corresponding note.
+
+**Step 3:** Finally, each `tagId` will be added to note through `NoteManager`.
+
+This whole process has executed the `TagNoteCommand`.
+
+=== [Proposed] Note filtering feature
+==== Description
+
+The user can filter notes by providing a filter in the <<UserGuide#note-list, `note list`>> command.
+
+Types of filter:
+
+* No prefix - filter by String keyword
+* Prefix `#` - filter by hashtag
+* Prefix `@` - filter by contact
+* Prefix `%` - filter by group
+
+==== Proposed Implementation
+
+The note filter mechanism is facilitated by the `Filter` inner class in link:{repoURL}/src/main/java/tagline/logic/commands/note/ListNoteCommand.java[`ListNoteCommand`].
+It contains a String filter value and the enum `FilterType`.
+
+A `Filter` is generated by the `NoteFilterUtil` inner class in link:{repoURL}/src/main/java/tagline/logic/parser/note/ListNoteParser.java[`ListNoteParser`] and passed into `ListNoteCommand`.
+
+===== Filter by String keyword
+
+Filter by keyword also implements the  link:{repoURL}/src/main/java/tagline/model/note/NoteContainsKeywordsPredicate.java[`NoteContainsKeywordsPredicate`]. The `Predicate` is passed into `Model#updateFilteredNoteList()` to list only notes that contain the keywords.
+
+Given below is an example scenario where the user enters a command to filter notes by keywords.
+
+**Step 1:** The user command is passed through the `LogicManager` and a `ListNoteCommand` with the `Filter` containing the keywords and `FilterType.KEYWORD` is returned.
+
+.Sequence diagram of parsing note filter user command to obtain a `ListNoteCommand`
+image::FilterKeywordSequenceDiagram1.png[]
+
+**Step 2:** The `ListNoteCommand` returned will be executed by the `LogicManager`. If a `Filter` exists and is of `FilterType.KEYWORD`, `ListNoteCommand#filterAndListByKeyword()` will be called.
+
+.Sequence diagram of executing `ListNoteCommand` to update filtered note list by keyword in `Model`
+image::FilterKeywordSequenceDiagram2.png[]
+
+The method will create a `NoteContainsKeywordsPredicate` and update the list of notes to be displayed via `Model`.
+
+image::FilterKeywordExample.png[]
```

Comment:
<img width="701" alt="Screenshot 2019-10-25 at 2 33 43 PM" src="https://user-images.githubusercontent.com/30684353/67548483-826c3380-f734-11e9-8bcc-85b8c4c7d5d2.png">

Error loading in this example diagram.